mod-permissions/src/main/resources/domain/mod-permissions/features/permissions.feature

```diff
@@ -0,0 +1,397 @@
+    # above, but why not test out this route and validate the permissions user schema too.
+    Given path 'perms/users/', permissionsUserId
+    And headers karate.merge(commonHeaders, optionsHeaders)
+    When method OPTIONS
+    Then status 204
+    And match header access-control-allow-methods contains "GET"
+  
+    Given path 'perms/users/', permissionsUserId
+    And headers commonHeaders
+    When method GET
+    Then status 200
+    # Validate response -- the PermissionUser object's schema.
+    And match response == 
+    """
+    {
+      "id": "#uuid",
+      "userId": "#uuid",
+      "permissions": "#array",
+      "metadata": "#object"
+    }
+    """
+    * def currentPerms = response.permissions
+
+    # Update the permissions for the given user.
+    * def newPerms = ["users.all", "login.all"]
+    * def permissionsToUpdate = karate.append(currentPerms, newPerms)
+    Given path 'perms/users/', permissionsUserId
+    And headers karate.merge(commonHeaders, optionsHeaders)
+    When method OPTIONS
+    Then status 204
+    And match header access-control-allow-methods contains "PUT"
+
+    Given path 'perms/users/', permissionsUserId
+    And headers commonHeaders
+    And request
+    """
+    {
+      "userId": "#(testUserId)",
+      "permissions": #(permissionsToUpdate)
+    }
+    """
+    When method PUT
+    Then status 200
+    And match response.permissions contains $currentPerms[*].name
+    And match response.permissions contains $newPerms[*].name
+
+    # Check that the permission user id is present in a permission's grantedTo property now that the
+    # user has been granted the permission.
+    Given path 'perms/permissions'
+    And param query = 'permissionName=="users.all"'
+    And headers commonHeaders
+    When method GET
+    Then status 200
+    And match response.permissions[0].grantedTo contains permissionsUserId
+
+Scenario: Get the permissions a user has, add a new permission, and remove one
+  Given path 'perms/users/', permissionsUserId, 'permissions'
+  And headers karate.merge(commonHeaders, optionsHeaders)
+  When method OPTIONS
+  Then status 204
+  And match header access-control-allow-methods contains "GET"
+
+  Given path 'perms/users/', permissionsUserId, 'permissions'
+  And headers commonHeaders
+  When method GET
+  Then status 200
+  And match response == { permissionNames: #array, totalRecords: #number }
+
+  # Add a permission to the user.
+  * def permToAdd = "configuration.all"
+  Given path 'perms/users/', permissionsUserId, 'permissions'
+  And headers karate.merge(commonHeaders, optionsHeaders)
+  When method OPTIONS
+  Then status 204
+  And match header access-control-allow-methods contains "POST"
+
+  Given path 'perms/users/', permissionsUserId, 'permissions'
+  And headers commonHeaders
+  And request { permissionName: #(permToAdd) }
+  When method POST
+  Then status 200
+  And match response == { permissionName: #string }
```

Comment:
Isn't this string check already covered by the following line?

Reply:
Yes the type check is a bit redundant.